homeassistant/util/__init__.py

```diff
@@ -260,25 +285,21 @@ def wrapper(*args, **kwargs):
             else:
                 host = args[0] if args else wrapper
 
-            if not hasattr(host, '_throttle_lock'):
-                host._throttle_lock = threading.Lock()
+            if not hasattr(host, '_throttle'):
+                host._throttle = {}
 
-            if not host._throttle_lock.acquire(False):
-                return None
+            if not id(self) in host._throttle:
+                host._throttle[id(self)] = ThrottleItem()
```

Comment:
namedtuple might be a good compromise that does not worsen the readability as much as a list and probably has a lower overhead than an object (and even less than a list maybe?).